drivers/spi/spi_ambiq.c

```diff
@@ -131,51 +131,31 @@ static int spi_ambiq_xfer(const struct device *dev, const struct spi_config *con
 
 	am_hal_iom_transfer_t trans = {0};
 
-	if (ctx->tx_len) {
-		trans.ui64Instr = *ctx->tx_buf;
-		trans.ui32InstrLen = 1;
-		spi_context_update_tx(ctx, 1, 1);
-
-		if (ctx->rx_buf != NULL) {
-			if (ctx->tx_len > 0) {
-				/* The instruction length can only be 0~5. */
-				if (ctx->tx_len > 4) {
-					spi_context_complete(ctx, dev, 0);
-					return -ENOTSUP;
-				}
-
-				/* Put the remaining TX data in instruction. */
-				trans.ui32InstrLen += ctx->tx_len;
-				for (int i = 0; i < trans.ui32InstrLen - 1; i++) {
-					trans.ui64Instr = (trans.ui64Instr << 8) | (*ctx->tx_buf);
-					spi_context_update_tx(ctx, 1, 1);
-				}
-			}
+	do {
+		trans.pui32TxBuffer = trans.pui32RxBuffer = NULL;
+		trans.eDirection = AM_HAL_IOM_FULLDUPLEX;
 
-			/* Set RX direction and hold CS to continue to receive data. */
-			trans.eDirection = AM_HAL_IOM_RX;
-			trans.bContinue = true;
+		if (ctx->tx_buf) {
+			trans.pui32TxBuffer = (uint32_t *)ctx->tx_buf;
+			trans.pui32RxBuffer = (uint32_t *)ctx->tx_buf;
```

Comment:
Why does this go in the RX buffer also?

Reply:
It's kind of a hack since in order to use full duplex you need to have a send and receive buffer but when only doing tx there will be no receive buffer provided.